src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,125 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
+    const queryKey = [
+        `validationRules`,
+        {
+            params: {
+                dataSet: datasetId,
+                fields: [
+                    'id',
+                    'importance',
+                    'operator',
+                    'leftSide',
+                    'rightSide',
+                    'displayInstruction',
+                    'displayDescription',
+                    'displayName',
+                ],
+            },
+        },
+    ]
+
+    return queryKey
+}
+
+const getValidationQueryKey = (
+    datasetId,
+    { periodId, orgUnitId, categoryComboId, categoryOptionIds }
+) => {
+    const params = {
+        pe: periodId,
+        ou: orgUnitId,
+        cc: categoryComboId,
+        cp: categoryOptionIds?.join(';'),
+    }
+
+    const validationQueryKey = [
+        `validation/dataSet/${datasetId}`,
+        {
+            params,
+        },
+    ]
+    return validationQueryKey
+}
+
+/** Returns validation rule violations grouped by importance */
+export const useValidationResult = () => {
+    const [datasetId] = useDataSetId()
+
+    const [periodId] = usePeriodId()
+    const [orgUnitId] = useOrgUnitId()
+    const {
+        attributeCombo: categoryComboId,
+        attributeOptions: categoryOptionIds,
+    } = useApiAttributeParams()
+
+    const validationQueryKey = getValidationQueryKey(datasetId, {
+        periodId,
+        orgUnitId,
+        categoryComboId,
+        categoryOptionIds,
+    })
+    const validationMetaDataQueryKey = getValidationMetaDataQueryKey(datasetId)
+
+    const activeMutations = useIsMutating({ mutationKey: validationQueryKey })
+    const enabled = activeMutations === 0
+
+    const results = useQueries({
+        queries: [
+            { queryKey: validationQueryKey, enabled },
+            { queryKey: validationMetaDataQueryKey, enabled },
+        ],
+    })
+
+    const refetch = useCallback(() => {
+        results.forEach((result) => result.refetch())
+    }, [results])
+
+    const [validationQuery, metaDataQuery] = results
+
+    return {
+        refetch,
+        error: validationQuery.error || metaDataQuery.error,
+        isLoading: validationQuery.isLoading || metaDataQuery.isLoading,
+        isRefetching:
+            validationQuery.isRefetching || metaDataQuery.isRefetching,
+        data:
+            validationQuery.isSuccess && metaDataQuery.isSuccess
+                ? groupValidationRuleViolationsByImportance(
+                      validationQuery.data,
+                      metaDataQuery.data
+                  )
+                : undefined,
```

Comment:
We could use stale data while refetching. Not sure if that'd have an impact on the UI though, so just a though I have:

```suggestion
            validationQuery.data && metaDataQuery.data
                ? groupValidationRuleViolationsByImportance(
                      validationQuery.data,
                      metaDataQuery.data
                  )
                : undefined,
```

Reply:
in [the design specs](https://docs.google.com/document/d/1vS1urvJRePRugFqrPkfOEjpkbij1AfTz0BYp6RKV8LU/edit#heading=h.padhp6i77jlc), we're hiding data while refetching (I think that makes sense UX-wise)

Reply:
:ok_hand: